protocols/raft/src/main/java/io/atomix/protocols/raft/roles/FollowerRole.java

```diff
@@ -237,8 +204,23 @@ protected VoteResponse handleVote(VoteRequest request) {
 
   @Override
   public void onLeaderHeartbeat(LeaderHeartbeatRequest request) {
+    raft.checkHeartbeatThread();
     logRequest(request);
 
+    final long currentTerm = raft.getTerm();
+    final RaftMember currentLeader = raft.getLeader();
```

Comment:
I think there is a (rare) chance for race condition here. Heartbeat thread reading them while raft thread update. But I think it will be harmless even if it gets inconsistent values. If it happens, probably the following log reports leader mismatch wrongly.

Reply:
Yes, you're right, and it might cause us to reject heartbeats unnecessarily, or accept a few when we shouldn't; but assuming we receive heartbeats fairly often, it should stabilize, I think.